services/src/main/java/org/keycloak/services/resources/admin/ClientRoleMappingsResource.java

```diff
@@ -134,7 +134,7 @@ public ClientRoleMappingsResource(UriInfo uriInfo, KeycloakSession session, Real
 
         return client.getRolesStream()
                 .filter(auth.roles()::canMapRole)
-                .filter(((Predicate<RoleModel>) user::hasRole).negate())
+                .filter(((Predicate<RoleModel>) user::hasDirectRole).negate())
```

Comment:
UX-wise admins should now see the client role available even though it is effectively granted, right?

Wondering if makes sense to update the hint for available roles to state that these roles are not directly granted to the user even though they might be shown as effective.

Reply:
@pedroigor Exactly, all roles which are not directly granted will be shown as available. Currently, for client roles, this hint is shown: "Assignable roles from this client". We can add a hint like "Contains effectively assigned roles which are not directly assigned." But we could only do this for English (and German), because we don't speak the other supported languages :(

Reply:
Works for me.